sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkDateTimeUtils.scala

```diff
@@ -46,7 +46,12 @@ trait SparkDateTimeUtils {
     // To support the (+|-)hh:m format because it was supported before Spark 3.0.
     formattedZoneId = singleMinuteTz.matcher(formattedZoneId).replaceFirst("$1$2:0$3")
 
-    ZoneId.of(formattedZoneId, ZoneId.SHORT_IDS)
+    try {
+      ZoneId.of(formattedZoneId, ZoneId.SHORT_IDS)
```

Comment:
Should we put the 2 lines above to `try` as well?

Reply:
Just adjusted comments. Thanks!